win32/tests/file_win32_ut/file_win32_ut.c

```diff
@@ -0,0 +1,952 @@
+
+    ///assert
+    ASSERT_IS_NULL(return_val);
+}
+
+/*Tests_SRS_FILE_43_002: [ If full_file_name is NULL then file_create shall fail and return NULL. ]*/
+/*Tests_SRS_FILE_WIN32_43_048: [ If full_file_name is NULL then file_create shall fail and return NULL. ]*/
+TEST_FUNCTION(file_create_fails_on_null_full_file_name)
+{
+    ///arrange
+    EXECUTION_ENGINE_HANDLE execution_engine = execution_engine_create(NULL);
+    ///act
+    FILE_HANDLE return_val = file_create(execution_engine, NULL, NULL, NULL);
+    ///assert
+    ASSERT_IS_NULL(return_val);
+}
+
+/*Tests_SRS_FILE_43_037: [ If full_file_name is an empty string, file_create shall fail and return NULL. ]*/
+/*Tests_SRS_FILE_WIN32_43_059: [ If full_file_name is an empty string, file_create shall fail and return NULL. ]*/
+TEST_FUNCTION(file_create_fails_on_empty_full_file_name)
+{
+    ///arrange
+    EXECUTION_ENGINE_HANDLE execution_engine = execution_engine_create(NULL);
+    ///act
+    FILE_HANDLE return_val = file_create(execution_engine, "", NULL, NULL);
+    ///assert
+    ASSERT_IS_NULL(return_val);
+}
+
+/*Tests_SRS_FILE_WIN32_43_041: [ file_create shall allocate a FILE_HANDLE. ]*/
+/*Tests_SRS_FILE_WIN32_43_001: [ file_create shall call CreateFileA with full_file_name as lpFileName, GENERIC_READ|GENERIC_WRITE as dwDesiredAccess, FILE_SHARED_READ as dwShareMode, NULL as lpSecurityAttributes, OPEN_ALWAYS as dwCreationDisposition, FILE_FLAG_OVERLAPPED|FILE_FLAG_WRITE_THROUGH as dwFlagsAndAttributes and NULL as hTemplateFile. ]*/
+/*Tests_SRS_FILE_WIN32_43_002: [ file_create shall call SetFileCompletionNotificationModes to disable calling the completion port when an async operations finishes synchrounously. ]*/
+/*Tests_SRS_FILE_WIN32_43_003: [ file_create shall initialize a threadpool environment by calling InitializeThreadpolEnvironment. ]*/
+/*Tests_SRS_FILE_WIN32_43_004: [ file_create shall obtain a PTP_POOL struct by calling execution_engine_win32_get_threadpool on execution_engine. ]*/
+/*Tests_SRS_FILE_WIN32_43_005: [ file_create shall register the threadpool environment by calling SetThreadpoolCallbackPool on the initialized threadpool environment and the obtained ptp_pool ]*/
+/*Tests_SRS_FILE_WIN32_43_006: [ file_create shall create a cleanup group by calling CreateThreadpoolCleanupGroup. ]*/
+/*Tests_SRS_FILE_WIN32_43_007: [ file_create shall register the cleanup group with the threadpool environment by calling SetThreadpoolCallbackCleanupGroup. ]*/
+/*Tests_SRS_FILE_WIN32_43_033: [ file_create shall create a threadpool io with the allocated FILE_HANDLE and on_file_io_complete_win32 as a callback by calling CreateThreadpoolIo ]*/
+/*Tests_SRS_FILE_WIN32_43_009: [ file_create shall succeed and return a non-NULL value. ]*/
+TEST_FUNCTION(file_create_succeeds)
+{
+    ///arrange
+    PTP_WIN32_IO_CALLBACK captured_callback;
+    ///act
+    FILE_HANDLE return_val = get_file_handle("file_create_succeeds.txt", &captured_callback);

```

Comment:
>get_file_handle [](start = 29, length = 15)

This should actually directly call `file_create`. You may need to refactor out the expectations from `get_file_handle` into a `setup_create_file_expectations` function.